src/entity.ts

```diff
@@ -855,6 +855,7 @@ export namespace entity {
       if (
         firstPathPartIsArray &&
         // check also if the property in question is actually an array value.
+        typeof entity.properties![firstPathPart] !== 'undefined' &&
```

Comment:
curious: why the check used is `typeof entity.properties![firstPathPart] !== 'undefined'` and not `entity.properties![firstPathPart] !== undefined` ?

Reply:
This change causes the test called `should encode a request when using a wildcard` to fail with `TypeError: Cannot read properties of undefined (reading 'arrayValue')`. I guess `typeof entity.properties![firstPathPart]` must evaluate to a string.

Reply:
Nevermind, I forgot to remove typeof as per your comment. This change is fine.

Reply:
I changed this everywhere and refactored this change out into a variable.

Reply:
looks great 👍